operator/v1/types_ingress.go

```diff
@@ -1399,6 +1399,132 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// actions specifies options for performing action on headers.
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set or deleted for TLS passthrough
+	// connections.
+	// Setting HSTS header is supported via another mechanism in Ingress.Spec.RequiredHSTSPolicies.
+	// The header settings made here can replace settings made in: cache-control, spec.clientTLS,
+	// spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId,
+	// and spec.httpHeaders.headerNameCaseAdjustments.
+	// The header settings made here can be overwritten by spec.httpHeaders or annotations
+	// on Route objects.
+	// Setting httpCaptureHeaders for the headers set via this API will not work.
+	// Setting "cookie" or "set-cookie" headers that conflict with OpenShift router's session cookies will result in undefined behavior.
+	// Note that the total size of all added headers *after* interpolating dynamic values
+	// must not exceed the value of spec.tuningOptions.headerBufferMaxRewriteBytes on the
+	// IngressController.
```

Comment:
It doesn't matter about whether they were added or not, it includes all headers modified or not right?
```suggestion
	// Note that the total size of all headers *after* interpolating dynamic values
	// must not exceed the value of spec.tuningOptions.headerBufferMaxRewriteBytes on the
	// IngressController.
```

Reply:
I think it is more accurate to say "added headers" (or possibly "net added headers") must not exceed `headerBufferMaxRewriteBytes`.  

If you want to get into the details, we have two settings:

* `spec.tuningOptions.headerBufferBytes` is the size of the buffer for headers.
* `spec.tuningOptions.headerBufferMaxRewriteBytes` is the reserved amount of the buffer.

An incoming request cannot exceed `headerBufferBytes` - `headerBufferMaxRewriteBytes`.  Net added headers cannot exceed `headerBufferMaxRewriteBytes`.  I think you can add more than `headerBufferMaxRewriteBytes` if you delete some of the headers from the original request.  

Maybe it would be best to refer the reader to the godoc for `headerBufferBytes` and `headerBufferMaxRewriteBytes` for the details.

Reply:
Ahh right I think I missed the difference between those two with this comment and was expecting the behaviour of `headerBufferBytes`. If that's the way this works then yes, maybe `net added` would be better and a reference to the other godoc to help explain the concepts would be helpful

Reply:
ok added.